From 1ce88613e6438f0ab9f50b826929f02408eb8f50 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Wed, 4 Jul 2018 08:22:12 +0200
Subject: [PATCH] device: disable rp_filter handling

Don't change rp_filter in any way, like in previous RHEL 7 releases.
See also https://bugzilla.redhat.com/show_bug.cgi?id=1492472.

https://bugzilla.redhat.com/show_bug.cgi?id=1593194
---
 src/devices/nm-device.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/devices/nm-device.c b/src/devices/nm-device.c
index 613e87034..ac9e1da08 100644
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -11440,7 +11440,7 @@ nm_device_set_ip_config (NMDevice *self,
 			priv->needs_ip6_subnet = FALSE;
 	}
 
-	if (IS_IPv4) {
+	if (IS_IPv4 && FALSE /* disabled on RHEL */) {
 		if (!nm_device_sys_iface_state_is_external_or_assume (self))
 			ip4_rp_filter_update (self);
 	}
@@ -12329,7 +12329,7 @@ queued_ip_config_change (NMDevice *self, int addr_family)
 
 	set_unmanaged_external_down (self, TRUE);
 
-	if (IS_IPv4) {
+	if (IS_IPv4 && FALSE /* disabled on RHEL */) {
 		if (!nm_device_sys_iface_state_is_external_or_assume (self)) {
 			priv->v4_has_shadowed_routes = _v4_has_shadowed_routes_detect (self);;
 			ip4_rp_filter_update (self);
-- 
2.17.0

